packages/tangle/scheduler.go

```diff
@@ -331,14 +372,29 @@ func (s *Scheduler) schedule() *Message {
 		msg := q.Front()
 		// a message can be scheduled, if it is ready
 		// (its issuing time is not in the future and all of its parents are eligible).
-		if msg != nil && !clock.SyncedTime().Before(msg.IssuingTime()) {
-			// compute how often the deficit needs to be incremented until the message can be scheduled
-			remainingDeficit := math.Dim(float64(msg.Size()), s.getDeficit(q.NodeID()))
-			nodeMana := s.accessManaCache.GetCachedMana(q.NodeID())
-			// find the first node that will be allowed to schedule a message
-			if r := int(math.Ceil(remainingDeficit / nodeMana)); r < rounds {
-				rounds = r
-				schedulingNode = q
+		// while loop to skip all the confirmed messages
+		foundMsg := false
+		for msg != nil && !clock.SyncedTime().Before(msg.IssuingTime()) && !foundMsg {
+			msgID, _, err := MessageIDFromBytes(msg.IDBytes())
+			if err != nil {
+				panic("MessageID could not be parsed!")
+			}
+			if s.tangle.ConfirmationOracle.IsMessageConfirmed(msgID) && clock.Since(msg.IssuingTime()) > time.Minute {
```

Comment:
same here, better to use a parameter for the time threshold

Reply:
Sure thing! Will replace it.